osidb/models/affect.py

```diff
@@ -483,6 +493,20 @@ def _validate_purl(self, **kwargs):
         except Exception as exc:
             raise ValidationError(f"Invalid purl '{self.purl}': {exc}")
 
+    def _validate_purl_and_ps_component(self, **kwargs):
+        """
+        Checks that ps_component matches the one included in purl if purl exists.
+        """
+        if (
+            self.purl
+            and self.ps_component
+            and self.ps_component != PackageURL.from_string(self.purl).name
```

Comment:
In the case user provides both but the PURL is invalid the conversion here would result in an exception which would most probably be unclear to the user and without any affect identification. We should catch it and transform into a better one.

Reply:
I was asked to re-review but this does not seem to be addressed. I the PURL string is no parse-able the validation will produce an error message which is unrelated to what is being validated here

Reply:
I was not referring to the missing tests but to the non-caught PURL parsing exception in the condition